### PR TITLE
[#975] Fix collision pair generation to ensure ordering doesnt matter

### LINKED ISSUE #975: PreventCollision and Active Collision Types are generating collision pairs sometimes
### Steps to Reproduce

Add the `PreventCollision` type actor to the scene first then a `Active` collision type actor.

Possible fix https://github.com/excaliburjs/Excalibur/commit/2e0f2e7d5ed9ae8df3e9a08dc50c321f632bcdde

This appears to be the offending line of code in `Pair.ts`:

```typescript
if (actorB.collisionType === CollisionType.PreventCollision || actorB.isKilled()) { return false; }
```

### Expected Result
`PreventCollision` and `Active` collision types should never collide. 

### Actual Result
Collision pairs are being generated and postcollision events are firing erroneously.

### Environment
Excalibur @ v0.16.0

### Current Workaround
Currently this does not cause too much trouble, but it does generate pairs unecessarily no physics resoultion happens. The workaround is to inspect postcollision events before handling them.

### PR DESCRIPTION
**Pull request by eonarheim**
===:clipboard: PR Checklist :clipboard:===
- [x] :pushpin: issue exists in github for these changes 
- [x] :microscope: existing tests still pass
- [x] :triangular_ruler: new tests written and passing / old tests updated with new scenario(s)
- [x] :page_facing_up: changelog entry added (or not needed)

==================

Closes #975

## Changes:

- Checks both sides of the collision
- Handles edge case with kill
- Adds tests

### CODE REVIEW

File: src/spec/CollisionSpec.ts [hunk 1]
```diff
@@ -50,6 +50,91 @@ describe('A Collision', () => {
     expect(actor2Collision).toBe(1);
   });
 
+  it('should order of actors collision should not matter when an Active and Active Collision', () => {
```
Comment (kamranayub) [inline src/spec/CollisionSpec.ts]: I'd probably remove `should` from beginning of these tests as it reads weird
